servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -1146,14 +1146,13 @@ void TextureStorage::texture_3d_update(RID p_texture, const Vector<Ref<Image>> &
 	{
 		Vector<Ref<Image>> images;
 		uint32_t all_data_size = 0;
-		images.resize(p_data.size());
 		for (int i = 0; i < p_data.size(); i++) {
 			Ref<Image> image = p_data[i];
 			if (image->get_format() != tex->validated_format) {
 				image = image->duplicate();
 				image->convert(tex->validated_format);
 			}
-			all_data_size += images[i]->get_data().size();
+			all_data_size += image->get_data().size();
 			images.push_back(image);
```

Comment:
```suggestion
			images[i] = image;
```
instead of removing the `resize` above avoids having to incrementally grow the vector.